### PR TITLE
Issue/2718 site search selection

### PR DESCRIPTION
**Pull request by kwonye**
Two things:

1. Just clean up that @aforcier mentioned that didn't make it into the previous merge. Mostly removing unused parameters

2. Addresses issue #2817. `getCurrentBlog()` currently checks for blog visibility. With the site picker search, the user can select a non-visible blog. Clicking 'View Site' does not work for non-visible blogs. So there are two options:

  1. Make any blog selected visible.
  2. getCurrentBlog() doesn't care about visibility.

My vote would be 2. Visibility seems to be a purely UI-driven concept, and auditing the calls to it doesn't yield any concerns. However, @maxme suggested 1. Thoughts? 

@roundhill @nbradbury @tonyr59h @aforcier

### CODE REVIEW

**Comment by daniloercoli** [comment]
> Make any blog selected visible.
getCurrentBlog() doesn't care about visibility.
My vote would be 2. Visibility seems to be a purely UI-driven concept, and auditing the calls to it doesn't yield any concerns. However, @maxme suggested 1. Thoughts?

My personal experience with the site picker seems to suggest the option #1. When I searched for a site is nice to have it available in the list right after.

**Comment by daniloercoli** [comment]
The code looks OK, and can be merged if we go with the option 2.

**Comment by nbradbury** [comment]
I think either option works, but my vote is for #1 since it seems more user-friendly. Either way, I think the search results should show hidden sites using the same "muted" styling I added [here](https://github.com/wordpress-mobile/WordPress-Android/blob/feature/2718-site-search/WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java#L150).

**Comment by kwonye** [comment]
Ready for review. Now when a hidden site is selected, it becomes visible.

File: WordPress/src/main/java/org/wordpress/android/WordPress.java [hunk 1]
```diff
@@ -446,6 +438,15 @@ public static Blog setCurrentBlog(int id) {
         return currentBlog;
     }
 
+    public static Blog setCurrentBlogAndSetVisible(int id) {
+        Blog blog = getBlog(id);
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/WordPress.java]: In the current version of the code `setCurrentBlogAndSetVisible` and `setCurrentBlog` despite doing almost the same thing they have different code. 
I also noticed that `setCurrentBlog` is very similar to `getBlog`.

My .2$: It's probably better to have `setCurrentBlogAndSetVisible` calling `setCurrentBlog`, and in order, change `setCurrentBlog` that should call `getBlog`.

Comment (kwonye) [inline WordPress/src/main/java/org/wordpress/android/WordPress.java]: > I also noticed that setCurrentBlog is very similar to getBlog.

Good catch! Didn't notice that. In fact, `setBlog()` returning a value didn't make sense so I looked at its usages and no one used the return value so I removed it.

>  It's probably better to have setCurrentBlogAndSetVisible calling setCurrentBlog and in order, change setCurrentBlog that should call getBlog.

So it was calling it before, but as the return value. Now, it calls it first and then sets the blog as visible. I believe that's what you meant.

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**